RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift

```diff
@@ -49,10 +58,21 @@ public struct FullScreenOverlayCoordinator: Sendable, FeatureReducer {
 	public func reduce(into state: inout State, childAction: ChildAction) -> Effect<Action> {
 		switch childAction {
 		case .root(.claimWallet(.delegate)):
-			.send(.delegate(.dismiss))
+			return .send(.delegate(.dismiss))
+
+		case let .root(.relinkConnector(.delegate(.newConnection(connectedClient)))):
+			userDefaults.setShowRelinkConnectorsAfterProfileRestore(false)
+			userDefaults.setShowRelinkConnectorsAfterUpdate(false)
```

Comment:
I can see this easily growing and ending up with this coordinator (which should be business logic independent, and only focused on presenting something anywhere on the app), into a class that takes care of very specific logic (as the one added now) for each case that it could present.

The ideal solution would be to have the `FullScreenOverlayCoordinator` be independent of `.claim` or `.relinkConnector`, and only take care of presenting a `FeatureView`. However, I don't know if there is an easy way of doing this with TCA. Have you looked into such approach?

Reply:
You’re right, although I don’t see any good solution right now. I already separated this logic from the `OverlayReducer` and in theory we should not overuse `OverlayWindowClient`, but maybe it’s worth looking for a better solution.